Translate  javascript-algorithms-and-data-structures-projects

LynnLian opened the pull request:
### 翻译章节

《算法和数据结构——实战》

### 变更文件

- [x] javascript-algorithms-and-data-structures-projects.json

- [ ] javascript-algorithms-and-data-structures-projects.md

### 翻译进度 

- [x] 60%

- [ ] 0%

Commit:
Progress: 1 / 22
id: 5a858944d96184f06fd60d61
title: Create Your First CSS Grid

- 翻译description，test

S1ngS1ng commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,14 +8,14 @@
       "description": [
-        "Turn any HTML element into a grid container by setting its <code>display</code> property to <code>grid</code>. This gives you the ability to use all the other properties associated with CSS Grid.",
-        "<strong>Note</strong><br>In CSS Grid, the parent element is referred to as the <dfn>container</dfn> and its children are called <dfn>items</dfn>.",
+        "为了令 HTML 元素包含在网格容器中，可将 <code>display</code> 的属性值设置为 <code>grid</code> 。此后，你将可以使用关于 CSS 的不同网格特性。",
```
这里似乎漏掉了 `any` 的翻译，应为 `任何 HTML 元素`。

另外，建议改一下语序：
`对于任何 HTML 元素，如果我们把它的 <code>display</code> 的属性值设置为 <code>grid</code>，那么它就成为了一个网格容器。`

---------

对于半角符号、数字、英文字母之后是全角符号的情况，则不需要加空格。

----------

`此后，你将可以使用关于 CSS 的不同网格特性。` 建议改成：
`然后，你就可以调用 CSS 网格的其他属性。`

S1ngS1ng commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,14 +8,14 @@
-        "Turn any HTML element into a grid container by setting its <code>display</code> property to <code>grid</code>. This gives you the ability to use all the other properties associated with CSS Grid.",
-        "<strong>Note</strong><br>In CSS Grid, the parent element is referred to as the <dfn>container</dfn> and its children are called <dfn>items</dfn>.",
+        "为了令 HTML 元素包含在网格容器中，可将 <code>display</code> 的属性值设置为 <code>grid</code> 。此后，你将可以使用关于 CSS 的不同网格特性。",
+        "<strong>敲黑板：</strong><br>在 CSS 网格中, 父元素可被视为 <dfn>container</dfn> 。那么他的子元素被称为  <dfn>items</dfn> 。",
```
“敲黑板” 很优秀 😂 不过为了整体统一，换成“注意” 吧。
请参考：https://github.com/FreeCodeCampChina/curriculum-cn/pull/12/files 最好能保证你们的翻译中，术语也是一致的

---------

请改一下这里的逗号，应当使用全角逗号。

--------------

`父元素可被视为 <dfn>container</dfn> 。那么他的子元素被称为  <dfn>items</dfn> 。` 建议改成：
`父元素叫做 <dfn>container</dfn>，而它的子元素叫 <dfn>items</dfn>。`

注意这里的 `而` 不表示转折关系

S1ngS1ng commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,14 +8,14 @@
+        "<strong>敲黑板：</strong><br>在 CSS 网格中, 父元素可被视为 <dfn>container</dfn> 。那么他的子元素被称为  <dfn>items</dfn> 。",
         "<hr>",
-        "Change the display of the div with the <code>container</code> class to <code>grid</code>."
+        "在包含以 <code>container</code> 为 class 的 div 中 ，把其中的 display 属性值设置为 grid。"
```
不要把 grid 外面的 `<code></code>` 删掉

---------

`在包含以 <code>container</code> 为 class 的 div 中 ，` =>
`在 class 为 <code>container</code> 的 div 中 ，`

S1ngS1ng commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,14 +8,14 @@
       "tests": [
         {
-          "text": "<code>container</code> class should have a <code>display</code> property with a value of <code>grid</code>.",
+          "text": "在名为 <code>container</code> 的 class 中，其 <code>display</code> 属性值应被设置为 <code>grid</code> 。",
```
类似上面的翻译，`class 为 <code>container</code> 的元素，其 <code>display</code> 属性值应被设置为 <code>grid</code> 。`

S1ngS1ng commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,14 +8,14 @@
         {
-          "text": "<code>container</code> class should have a <code>display</code> property with a value of <code>grid</code>.",
+          "text": "在名为 <code>container</code> 的 class 中，其 <code>display</code> 属性值应被设置为 <code>grid</code> 。",
           "testString": "assert(code.match(/.container\\s*?{[\\s\\S]*display\\s*?:\\s*?grid\\s*?;[\\s\\S]*}/gi), '<code>container</code> class should have a <code>display</code> property with a value of <code>grid</code>.');"
```
这里也是要翻译的噢，翻译第二个参数就好啦

S1ngS1ng reviewed (changes requested):
下次提交前，请先检查一下格式哈。包括标点，空格之类的

LynnLian commented:
以上格式及翻译问题已收到并修改。会多注意格式问题 :) 谢谢

Commit:
Progress: 1 / 22
id: 5a858944d96184f06fd60d61
title: Create Your First CSS Grid

- 修改了上一 pull request 的语法及标点问题。

S1ngS1ng commented:
+ @huluoyang @wudifeixue 　

Commit:
Progress: 2 / 22
id: 5a9036d038fddaf9a66b5d32
title: Add Columns with grid-template-columns

- 翻译description， test

huluoyang commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,15 +8,15 @@
       "description": [
-        "Turn any HTML element into a grid container by setting its <code>display</code> property to <code>grid</code>. This gives you the ability to use all the other properties associated with CSS Grid.",
-        "<strong>Note</strong><br>In CSS Grid, the parent element is referred to as the <dfn>container</dfn> and its children are called <dfn>items</dfn>.",
+        "对于任何 HTML 元素，如果我们把它的 <code>display</code> 的属性值设置为 <code>grid</code>，那么它就成为了一个网格容器。然后，你就可以调用 CSS 网格的其他属性。",
```
use => 使用
call => 调用

Commit:
按以下规则修改用词：
use => 使用
call => 调用

Commit:
算法和数据结构 - 项目实践 （1/5)

Palindrome Checker

- 翻译description，test

Commit:
Merge branch 'translate' of github.com:freeCodeCampChina/challenges into translate

Commit:
算法和数据结构 - 项目实践 （2/5)

Roman Numeral Converter

- 翻译description，test

huluoyang commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -8,15 +8,15 @@
       "description": [
-        "Turn any HTML element into a grid container by setting its <code>display</code> property to <code>grid</code>. This gives you the ability to use all the other properties associated with CSS Grid.",
-        "<strong>Note</strong><br>In CSS Grid, the parent element is referred to as the <dfn>container</dfn> and its children are called <dfn>items</dfn>.",
+        "对于任何 HTML 元素，如果我们把它的 <code>display</code> 的属性值设置为 <code>grid</code>，那么它就成为了一个网格容器。然后，你就可以使用 CSS 网格的其他属性。",
```
turn into  变成

huluoyang commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -64,17 +64,17 @@
+        "在 <code>grid-template-columns</code> 属性中，参数的个数代表列的个数，参数的值代表列的宽度。",
         "<hr>",
-        "Give the grid container three columns that are <code>100px</code> wide each."
+        "在容器中设置三个列，每列宽度为 <code>100px</code>。"
```
网格容器

huluoyang commented on 01-responsive-web-design/css-grid.json:
```diff
@@ -64,17 +64,17 @@
         {
-          "text": "<code>container</code> class should have a <code>grid-template-columns</code> property with three units of <code>100px</code>.",
-          "testString": "assert(code.match(/.container\\s*?{[\\s\\S]*grid-template-columns\\s*?:\\s*?100px\\s*?100px\\s*?100px\\s*?;[\\s\\S]*}/gi), '<code>container</code> class should have a <code>grid-template-columns</code> property with three units of <code>100px</code>.');"
+          "text": "在容器中应该有三个列，而且每列宽度为 <code>100px</code>。",
```
`在容器中应该有三个列，而且每列宽度为 <code>100px</code>。`
`容器有三列，每列宽度为 <code>100px</code>。`
下面这种表述是否简洁点。

LynnLian replied:
因为我现在没有再翻译网格css-grid.json这个文件，所以我就不修改了。

huluoyang commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
+        "<strong>注意</strong><br>检查回文时，你需要先除去 <strong>所有非字母数字的字符</strong> （标点、空格和符号）并且将所有字符转换成字母大写或字母小写。",
+        "我们将会传递不同格式的字符串，例如：<code>\"racecar\"</code>，<code>\"RaceCar\"</code>，<code>\"race CAR\"</code> 等等。",
+        "我们也会传递一些包含特殊符号的字符串，例如 <code>\"2A3*3a2\"</code>，<code>\"2A3  3a2\"</code> 和 <code>\"2_A3*3#A2\"</code>。",
+        "当你遇到困难时，请记得浏览 <a href=\"http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514\" target=\"_blank\">Read-Search-Ask</a>。然后，编写属于你自己的代码。"
```
use => 使用，而不是浏览，其他地方同理。

S1ngS1ng replied:
参考群内讨论，昨天微信群里讨论过这里应该如何翻译。
可以参考 [我的翻译](https://github.com/FreeCodeCampChina/challenges/pull/11/files#diff-10077b5d43aefc399fb6f9c6541a20d6R13)

huluoyang commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -102,117 +102,117 @@
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "把给定的数字转变为罗马数字。",
+        "所有为<a href=\"http://www.mathsisfun.com/roman-numerals.html\" target=\"_blank\">罗马数字</a>的答案都必须是字母大写。",
+        "当你遇到困难时，请记得浏览 <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a>。你可以尝试结伴编程，但请编写属于你自己的代码。 "
```
pair program  => 结对编程。 这里强调一对，而不是同伴，因为同伴可能很多。
`Try to pair program. Write your own code.`  尝试结对编程，书写自己的代码。
结对编程和书写自己的代码并不矛盾，所以`但`这种语气的变化没有必要，`请`更是可以忽略。
结对编程的要义是：一人编写，一人评注，两人协作，效率更快，彼此都有提升。

S1ngS1ng replied:
同上

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
-        "We'll pass strings with varying formats, such as <code>\"racecar\"</code>, <code>\"RaceCar\"</code>, and <code>\"race CAR\"</code> among others.",
-        "We'll also pass strings with special symbols, such as <code>\"2A3*3a2\"</code>, <code>\"2A3  3a2\"</code>, and <code>\"2_A3*3#A2\"</code>.",
-        "Remember to use <a href=\"http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514\" target=\"_blank\">Read-Search-Ask</a> if you get stuck. Write your own code."
+        "如果给定的一个字符串是回文，那么返回 true， 否则返回 false。",
```
`如果给定的一个字符串是回文` 建议 =>
`如果传入的字符串是回文的`

-------

请去掉逗号之后多余的空格

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
-        "Remember to use <a href=\"http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514\" target=\"_blank\">Read-Search-Ask</a> if you get stuck. Write your own code."
+        "如果给定的一个字符串是回文，那么返回 true， 否则返回 false。",
+        "<dfn>回文</dfn> 指在忽略标点符号、大小写和空格的前提下，将一个词语或句子按相反的顺序重新排列后，所得到的结果和原来的一样。",
+        "<strong>注意</strong><br>检查回文时，你需要先除去 <strong>所有非字母数字的字符</strong> （标点、空格和符号）并且将所有字符转换成字母大写或字母小写。",
```
`<br>` 表示新的一行，建议在 `注意` 之后加冒号，`注意：`  @huluoyang 可以添加到 guideline

------

@huluoyang 这里，`<strong></strong>` tag 两边应该不需要空格吧？在页面中显示的是 inline 的效果　

-------

`（标点、空格和符号）` => `，比如标点、空格和符号。`

--------

`并且将所有字符转换成字母大写或字母小写。` => 
`并且在判断前将字符串中的所有字符一律转换成大写或小写。`

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
+        "如果给定的一个字符串是回文，那么返回 true， 否则返回 false。",
+        "<dfn>回文</dfn> 指在忽略标点符号、大小写和空格的前提下，将一个词语或句子按相反的顺序重新排列后，所得到的结果和原来的一样。",
+        "<strong>注意</strong><br>检查回文时，你需要先除去 <strong>所有非字母数字的字符</strong> （标点、空格和符号）并且将所有字符转换成字母大写或字母小写。",
+        "我们将会传递不同格式的字符串，例如：<code>\"racecar\"</code>，<code>\"RaceCar\"</code>，<code>\"race CAR\"</code> 等等。",
```
`传递` => `传入`
下同

------

英文用逗号表示列举，中文用的是顿号。
下同

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -102,117 +102,117 @@
-        "Convert the given number into a roman numeral.",
-        "All <a href=\"http://www.mathsisfun.com/roman-numerals.html\" target=\"_blank\">roman numerals</a> answers should be provided in upper-case.",
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "把给定的数字转变为罗马数字。",
```
`给定` 建议 => `传入`
既然要写的是函数，个人觉得用 `传入` 更好

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -102,117 +102,117 @@
-        "All <a href=\"http://www.mathsisfun.com/roman-numerals.html\" target=\"_blank\">roman numerals</a> answers should be provided in upper-case.",
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "把给定的数字转变为罗马数字。",
+        "所有为<a href=\"http://www.mathsisfun.com/roman-numerals.html\" target=\"_blank\">罗马数字</a>的答案都必须是字母大写。",
```
其实可以直接说：
`转换后的罗马数字字母必须都是大写。`
或
~~`函数返回的结果必须都是大写。`~~

前一种方便加链接

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -102,117 +102,117 @@
-          "text": "<code>convertToRoman(2)</code> should return \"II\".",
-          "testString": "assert.deepEqual(convertToRoman(2), \"II\", '<code>convertToRoman(2)</code> should return \"II\".');"
+          "text": "<code>convertToRoman(2)</code> 应该返回 \"II\".",
+          "testString": "assert.deepEqual(convertToRoman(2), \"II\", '<code>convertToRoman(2)</code> 应该返回 \"II\".');"
```
这两行，要用中文句号。底下所有的都是，原文没有的请补上

S1ngS1ng reviewed (changes requested):
@huluoyang 
Pattern request：
1. `<strong>` `<dfn>` 等行内元素两边需不需要空格，`<a>` 呢？
2. 建议列出可复用的翻译。比如这道题目里的 Read-Search-Ask。翻译相关部分的其他人也应知晓

Commit:
Caesars Cipher

- 翻译description，test

（对于"cipher"，"shifted"的翻译拿捏不准，等候反馈。）

LynnLian commented:
因为翻译了第三小节才看到对于前两小节的修改，所以我先commit第三小节的翻译，接着再修改前两小节。

另外，以下问题会等候 @huluoyang 的回复后作全文档更新。
1. `<a>` 和 `<strong> <dfn>` 等行内元素两边需不需要空格
2. 可复用的翻译

@S1ngS1ng FYI

huluoyang commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -244,31 +244,31 @@
-        "Write a function which takes a <a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> encoded string as input and returns a decoded string.",
-        "All letters will be uppercase. Do not transform any non-alphabetic character (i.e. spaces, punctuation), but do pass them on.",
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "其中一个最广为人知而且简单的 <dfn>加密技术</dfn> 是 <code>凯撒密码</code>，也被称为 <code>变换加密</code>。在 <code>变换加密</code> 中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。",
```
Caesar cipher => 凯撒密码
shift cipher => 移位密码
ciphers => 密码
第一句应该翻译成：世界上最简单、最著名的密码是凯撒密码，也叫移位密码。

S1ngS1ng replied:
@huluoyang 我的理解是，`shift cipher` 是一个类，“移位密码”。所以应该说 `世界上最简单、最著名的密码是凯撒密码，它是移位密码中的一种。`

https://en.wikipedia.org/wiki/Caesar_cipher

注意 `shift cipher` 没有大写，所以这里 `shift cipher` 不是专有名词。the 的用法符合这一条：

![image](https://user-images.githubusercontent.com/13592559/43154948-d08b961e-8f2a-11e8-95ce-ca5ed8227bb9.png)

shift cipher 是 substitution cipher 中的一种实现。参考：https://en.wikipedia.org/wiki/Substitution_cipher

LynnLian replied:
@S1ngS1ng @huluoyang 
请问对于：

**凯撒密码就是移位密码。**

还是

**凯撒密码是移位密码的一种。**

这个讨论有最终结果吗？

huluoyang commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -244,31 +244,31 @@
-        "All letters will be uppercase. Do not transform any non-alphabetic character (i.e. spaces, punctuation), but do pass them on.",
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "其中一个最广为人知而且简单的 <dfn>加密技术</dfn> 是 <code>凯撒密码</code>，也被称为 <code>变换加密</code>。在 <code>变换加密</code> 中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。",
+        "<a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> 是一个被广泛使用的加密技术，明文中的所有字母都被移动 13 位。因此，'A' &harr; 'N', 'B' &harr;  'O' 等等。",
```
decode => 解码
encode => 编码
encrypt => 加密
decrypt => 解密

S1ngS1ng commented:
@LynnLian inline elements 理论上可以不加空格。但这个不应当与“全半角之间添加空格”的原则冲突：
- 错误：`请点击<a>help doc</a>获取帮助`
- 正确：`请点击 <a>help doc</a> 获取帮助`
- 正确：`请点击<a>帮助文档</a>获取帮助`

@huluoyang Thoughts?

Commit:
Merge branch 'translate' of https://github.com/freeCodeCampChina/challenges into translate

Commit:
修改1，2，3部分在pr的comment。

Commit:
Telephone Number Validator

- 翻译description，test。

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -244,31 +244,31 @@
+        "世界上最简单、最著名的<dfn>密码</dfn>是<code>凯撒密码</code>，也叫<code>移位密码</code>。在<code>移位密码</code>中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。",
+        "<a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> 是一个被广泛使用的编码技术，明文中的所有字母都被移动 13 位。因此，'A' &harr; 'N', 'B' &harr;  'O' 等等。",
+        "请编写一个函数，用于解码一个被 <a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> 编码的字符串，然后返回解码后的结果。",
+        "所有解码后的字母都必须为字母大写。请不要解码非字母的字符（例如，空格、标点符号），但是你需要转递他们到结果中。",
```
`但是你需要转递他们到结果中。` => `但你需要在结果中保留它们`

S1ngS1ng commented:
@LynnLian 如果你没有在翻译 `css-grid` 那个文件，那为什么你的 changes 里会有那个文件的改动？
https://github.com/FreeCodeCampChina/challenges/pull/8/commits
你的 PR 里的前三个 commits 就是那些改动。如果你不需要，请删除。
如果不确定怎么删除，有空了联系我

S1ngS1ng committed:
Add .gitignore, including patterns for OS, Vim, VSCode, Jetbrains and (#14)

this's good.

Commit:
revert commit f78814b6c5cf1c70992e629531065b256eb95bca

S1ngS1ng committed:
Translation of intermediate-algorithm-scripting, 50% (#11)

* Translation of intermediate-algorithm-scripting, first half

* Address comments

S1ngS1ng committed:
Add CONTRIBUTING.md (#15)

* Add CONTRIBUTING.md, add guide of git

* address review comments

- Pick pull --rebase among three options
- Update guideline of resolving conflicts

* Add git FAQ. Remove labels def as for not needed

* Fix style issue, using HTML tags

* Update Details w/ GFM, fix wrong command

* Update FAQ with guide of resolving conflicts and clarification of git add . vs git add -A

* Update CONTRIBUTING.md

更新了下排版

* Update CONTRIBUTING.md

更改符号的使用

* Update CONTRIBUTING.md

continue fix some type error

S1ngS1ng committed:
Update .gitignore that captures .vscode.*\.json (#28)

Add a pattern that captures .vscode.*\.json, as provided by #7

huluoyang committed:
Fix (#29)

* fix some typo error

* remove upcoming fileds

* resolve conflicts

S1ngS1ng committed:
(WIP) style-guide (#33)

Fix inline code in summary title that is not rendering

ifyour committed:
Translate managing-packages-with-npm (#1)

* translation 2/10

* translation 4/10

* translate assert & revise

* translation 2/10

* revise

* fix typo

* translation 2/10

* fix typo

* revise

* translate the document

* fix typo

* translate the title

* remove upcoming

XERO75 committed:
Translate responsive-web-design-projects (#2)

* part1

* update previous & 2/5 finished

* space fix

* fix part1&part2 spacing

stanleyyylau committed:
Translate information-security-with-helmet (#4)

* 完成helmet第一小节翻译

* Finish 10%

* Remove IDE generated files

* Update according to comments

XiaoLuo01 committed:
Translate object-oriented-programming (#12)

* finish 50% translation of object-oriented-programming

* modify translate

* change 对象 to  物体 in the real life.

focus-kfc committed:
Translate basic-css (#17)

* add: 完成BasicCSS的前4个教程。

* alter: 修改调整翻译中文。

* add: 继续翻译3小节。

* fixed: 修复语义，以及翻译格式。

* update: 替换空格&修改css class翻译。

* remove extra whitespace

stevending1st committed:
Css grid (#30)

* 01-responsive-web-design\css-grid.json is completed

* translatre Update 2018/7/23-11:00

* translate Updata 2018/7/23-11:10

* css-grid-cn.json updata no.2

* Del '-cn' files

* 'css-grid.json' Updata no.3

* css-grid.json updata no.4

* 'css-grid.md' Updata no.1

* css-grid.md updata no.2, .h空间’ changed to '间距'

* css-grid.json Updata change some punctuations

* 01-responsive-web-design\css-grid.json is completed

* translatre Update 2018/7/23-11:00

* translate Updata 2018/7/23-11:10

* css-grid-cn.json updata no.2

* Del '-cn' files

* css-grid.json updata no.4

* css-grid.json Updata change some punctuations

* css-grid.md updata 2018/7/27 22:13

rainpure committed:
responsive-web-design-principles translate (#35)

S1ngS1ng committed:
更新：翻译格式与建议 (#37)

* Update style guide, fixing format issue, adding anchor link in CONTRIBUTING.md and adding content

* Add em-dash style guide

* Add two more rules

S1ngS1ng commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -301,123 +301,123 @@
         "<blockquote>555-555-5555<br>(555)555-5555<br>(555) 555-5555<br>555 555 5555<br>5555555555<br>1 555 555 5555</blockquote>",
-        "For this challenge you will be presented with a string such as <code>800-692-7753</code> or <code>8oo-six427676;laskdjf</code>. Your job is to validate or reject the US phone number based on any combination of the formats provided above. The area code is required. If the country code is provided, you must confirm that the country code is <code>1</code>. Return <code>true</code> if the st ...
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "在这个挑战中，你将会看到例如 code>800-692-7753</code> 或者 <code>8oo-six427676;laskdjf</code> 的号码。你的任务是根据上面不同的格式组合，判断它是否美国号码。区号是必须的。如果提供国家代码，则必须确认国家代码为 <code>1</code>。如果这是有效的美国电话就返回 <code>true</code>，否则返回 <code>false</code>。",
```
这里的`<code>`标签错了

S1ngS1ng reviewed (changes requested):
上面提到的两个 comments 麻烦你改下。以及，根据最新的 style guide，记得把`<code>`两边的空格删掉

其他应该没问题了

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
-        "We'll pass strings with varying formats, such as <code>\"racecar\"</code>, <code>\"RaceCar\"</code>, and <code>\"race CAR\"</code> among others.",
-        "We'll also pass strings with special symbols, such as <code>\"2A3*3a2\"</code>, <code>\"2A3  3a2\"</code>, and <code>\"2_A3*3#A2\"</code>.",
-        "Remember to use <a href=\"http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514\" target=\"_blank\">Read-Search-Ask</a> if you get stuck. Write your own code."
+        "如果给定的一个字符串是回文，那么返回 true，否则返回 false。",
```
这里请不要删掉`<code>和</code>`，这在题目显示的时候是至关重要的（因为有红色高亮）。 
正确的写法应该是：  
`如果给定的一个字符串是回文，那么返回<code>true</code>，否则返回<code>false</code>。`

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -8,65 +8,65 @@
-        "We'll also pass strings with special symbols, such as <code>\"2A3*3a2\"</code>, <code>\"2A3  3a2\"</code>, and <code>\"2_A3*3#A2\"</code>.",
-        "Remember to use <a href=\"http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514\" target=\"_blank\">Read-Search-Ask</a> if you get stuck. Write your own code."
+        "如果给定的一个字符串是回文，那么返回 true，否则返回 false。",
+        "<dfn>回文</dfn> 指在忽略标点符号、大小写和空格的前提下，将一个词语或句子按相反的顺序重新排列后，所得到的结果和原来的一样。",
```
这里不妨留着`palindrome`这个单词，然后解释这就是回文。在做题的同时顺便学一下英语单词嘛。同时汉语这部分翻译的太难理解了，可以简化一下。
就是：
`<dfn>palindrome（回文）</dfn>，指在忽略标点符号、大小写和空格的前提下，正着读和反着读一模一样。`

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -244,31 +244,31 @@
-        "Write a function which takes a <a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> encoded string as input and returns a decoded string.",
-        "All letters will be uppercase. Do not transform any non-alphabetic character (i.e. spaces, punctuation), but do pass them on.",
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "世界上最简单、最著名的<dfn>密码</dfn>是<code>凯撒密码</code>，也叫<code>移位密码</code>。在<code>移位密码</code>中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。",
```
这点从上下文意思上来讲ciphers翻译成密码是错误的。会被误认为是password。  
正确的翻译方法因该是`加密方法`，或者说`加密技术`。

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -244,31 +244,31 @@
-        "Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck/19514' target='_blank'>Read-Search-Ask</a> if you get stuck. Try to pair program. Write your own code."
+        "世界上最简单、最著名的<dfn>密码</dfn>是<code>凯撒密码</code>，也叫<code>移位密码</code>。在<code>移位密码</code>中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。",
+        "<a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> 是一个被广泛使用的编码技术，明文中的所有字母都被移动 13 位。因此，'A' &harr; 'N', 'B' &harr;  'O' 等等。",
+        "请编写一个函数，用于解码一个被 <a href=\"https://en.wikipedia.org/wiki/ROT13\" target='_blank'>ROT13</a> 编码的字符串，然后返回解码后的结果。",
```
把ROT13的链接换成不被墙的地址吧：
http://www.baike.com/wiki/ROT13

LynnLian replied:
我以为敏感话题才会墙维基百科。。。
已修改 ：）

wudifeixue replied:
嘛，不管如何提前预防吧，而且默认英文的也是需要改的。

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -301,123 +301,123 @@
       "description": [
-        "Return <code>true</code> if the passed string looks like a valid US phone number.",
-        "The user may fill out the form field any way they choose as long as it has the format of a valid US number. The following are examples of valid formats for US numbers (refer to the tests below for other variants):",
+        "传入的字符串是美国电话号码格式就返回 <code>true</code>。",
```
valid没有被翻译，这里valid的含义还是挺重要的。
可以这么翻译：
`如果传入的字符串是一个有效的美国电话号码格式，则返回<code>true</code>。`

wudifeixue commented on 02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects.json:
```diff
@@ -301,123 +301,123 @@
-        "Return <code>true</code> if the passed string looks like a valid US phone number.",
-        "The user may fill out the form field any way they choose as long as it has the format of a valid US number. The following are examples of valid formats for US numbers (refer to the tests below for other variants):",
+        "传入的字符串是美国电话号码格式就返回 <code>true</code>。",
+        "只要是美国电话号码的格式，用户可以按照他们的方式填写表单中的电话号码。以下是一些正确的例子（其他格式变形请参考以下例子）：",
```
同上问题，valid在翻译的时候含义丢失了。
可以这么翻译：  
`只要是有效的美国电话号码的格式，用户可以按照他们的方式填写表单中的电话号码。以下是一些正确的例子（其他格式变形请参考以下例子）：`

wudifeixue reviewed (changes requested):
总体来讲是很棒的！感谢你的贡献！

Commit:
修改：
1. 已翻译部分的comment；
2. 按style-guide修改<code>、<a>、<b>、<dfn>、<strong>的空格。

Commit:
Cash Register

- 翻译description和test

wudifeixue commented:
话说我看你[改的那部分](https://github.com/FreeCodeCampChina/challenges/pull/8/commits/c22c82ae6eed1aa30a3b19b2b77ec750419c368a)都挺好的，但是怎么一下多了那么多其他的commit呀？  
@S1ngS1ng 这个和 #6 是一样的情况吧？

S1ngS1ng commented:
@wudifeixue 对的。CONTRIBUTION.md 里面，FAQ 部分更新了解决方案。
@LynnLian 请参考 [常见问题](https://github.com/FreeCodeCampChina/challenges/blob/60ed482767dac448fc3be329dcb96ddcc7aba749/CONTRIBUTING.md#%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98) 的最后一条

LynnLian commented:
@S1ngS1ng @wudifeixue 
按指示开了新的[pr](https://github.com/FreeCodeCampChina/challenges/pull/48)。请查看，谢谢。

S1ngS1ng commented:
Changes have been addressed in #48
